src/Compilers/Core/Portable/SourceGeneration/PendingEdit.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "In progress")]
+    public abstract class PendingEdit
+    {
+        internal abstract GeneratorDriverState Commit(GeneratorDriverState state);
+
+        internal abstract bool AcceptedBy(ISourceGenerator generator);
+
+        internal abstract UpdateContext TryApply(ISourceGenerator generator, UpdateContext context);
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "In progress")]
+    public abstract class AdditionalFileEdit : PendingEdit
+    {
+        public class AddtionalFileAddedEdit : AdditionalFileEdit
```

Comment:
```suggestion
        public sealed class AddtionalFileAddedEdit : AdditionalFileEdit
```

Reply:
Generally frown on types which are both `public` and nested. Is there a reason for not making this a top level or non-public type?